code/datums/gamemode/role/cultist.dm

```diff
@@ -3,38 +3,60 @@
 	name = "Cultist"
 	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Chaplain", "Head of Personnel", "Internal Affairs Agent")
 	logo_state = "cult-logo"
+	greets = list("default","custom","roundstart","admintoggle")
 
+/datum/role/cultist/New(var/datum/mind/M, var/datum/faction/fac=null, var/new_id)
+	..()
+	wikiroute = role_wiki[ROLE_CULTIST]
 
 /datum/role/cultist/OnPostSetup()
 	. = ..()
 	if(!.)
 		return
 
+	antag.current.add_language(LANGUAGE_CULT)
 	if(!(locate(/spell/cult) in antag.current.spell_list))
 		antag.current.add_spell(new /spell/cult/trace_rune, "cult_spell_ready", /obj/abstract/screen/movable/spell_master/bloodcult)
 		antag.current.add_spell(new /spell/cult/erase_rune, "cult_spell_ready", /obj/abstract/screen/movable/spell_master/bloodcult)
 
 /datum/role/cultist/RemoveFromRole(var/datum/mind/M)
+	antag.current.remove_language(LANGUAGE_CULT)
 	for(var/spell/cult/spell_to_remove in antag.current.spell_list)
 		antag.current.remove_spell(spell_to_remove)
 	..()
 
-/datum/role/cultist/Greet()
-	to_chat(antag.current, {"<span class='sinister'><font size=3>You are a cultist of <span class='danger'><font size=3>Nar-Sie</font></span>!</font><br>
-	I, the Geometer of Blood want you to thin the veil between reality and his realm<br>
-	so I can pull this place onto my plane of existence.<br>
-	You've managed to get a job here, and the time has come to put our plan into motion.<br>
-	However the veil is currently so thick that I can barely bestow any power to you.<br>
-	Other cultists made their way into the crew. Talk to them. <span class='danger'>Self Other Technology</span>!<br>
-	Meet up with them. Raise an altar in my name. <span class='danger'>Blood Technology Join</span>!<br>
-	</span>"})
-
-/*
-/datum/role/cultist/RoleTopic(href, href_list)
-	..()
-	if(!ismob(usr))
+/datum/role/cultist/Greet(var/greeting,var/custom)
+	if(!greeting)
 		return
-*/
+
+	var/icon/logo = icon('icons/logos.dmi', logo_state)
+	switch(greeting)
+		if ("roundstart")
+			to_chat(antag.current, {"<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='sinister'><font size=3>You are a cultist of <span class='danger'><font size=3>Nar-Sie</font></span>!</font><br>
+				I, the Geometer of Blood, want you to thin the veil between your reality and my realm<br>
+				so I can pull this place onto my plane of existence.<br>
+				You've managed to get a job here, and the time has come to put our plan into motion.<br>
```

Comment:
Needs some action works, right now it sounds like a peptalk, not enough drama.

Rather than 'managed to get a job here', why not 'infiltrated the crew'?